### PR TITLE
Moved and revised the bool article

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #10726 
Supported by dotnet/samples#1796

### CODE REVIEW

File: docs/csharp/toc.yml [hunk 1]
```diff
@@ -1318,9 +1318,11 @@
         - name: Built-in numeric conversions
           href: language-reference/builtin-types/numeric-conversions.md
         - name: bool
-          href: language-reference/keywords/bool.md
+          href: language-reference/builtin-types/bool.md
+          displayName: "Boolean, true, false"
         - name: char
           href: language-reference/builtin-types/char.md
+          displayName: "Unicode, character, UTF-16"
```
Comment (BillWagner) [inline docs/csharp/toc.yml]: nice!

File: docs/csharp/language-reference/builtin-types/bool.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Literals
+
+You can use the `true` and `false` literals to initialize a `bool` variable or to pass a `bool` value:
+
+[!code-csharp-interactive[bool literals](~/samples/csharp/language-reference/builtin-types/BoolType.cs#Literals)]
+
+## Conversions
+
+C# doesn't provide built-in conversions between the `bool` type and other types. However, .NET provides the methods that you can use to convert to or from the `bool` type. For more information, see the [Converting to and from Boolean values](/dotnet/api/system.boolean#converting-to-and-from-boolean-values) section of the <xref:System.Boolean?displayProperty=nameWithType> API reference page.
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/bool.md]: There's an inaccuracy here that I'm not sure how to phrase. There is an implicit identity nullable conversion from `bool` to `bool?`.  That's what's talked in the next section. Here's one suggestion, but I'm not thrilled with it.

```suggestion
The only built-in conversion involving `bool` is the implicit conversion from `bool` to `bool?`, discussed in the following section. .NET provides the methods that you can use to convert to or from the `bool` type. For more information, see the [Converting to and from Boolean values](/dotnet/api/system.boolean#converting-to-and-from-boolean-values) section of the <xref:System.Boolean?displayProperty=nameWithType> API reference page.
```

**Review by BillWagner — COMMENTED** [review]
This is great work @pkulikov 

I've got one concern in the wording of one area. I'd like your thoughts on better wording.

File: docs/csharp/language-reference/builtin-types/bool.md (hunk 2 above)
Comment (pkulikov) [reply docs/csharp/language-reference/builtin-types/bool.md]: Indeed, and then also explicit conversion from `bool?` back. I'm thinking about wording

Comment (pkulikov) [reply docs/csharp/language-reference/builtin-types/bool.md]: @BillWagner please take a look at the latest commit in which I've corrected the inaccuracy.

**Review by BillWagner — APPROVED** [review]
This is great @pkulikov 

I'll :shipit: now.

**Pull request merged**